sql/core/src/main/scala/org/apache/spark/sql/internal/HiveSerDe.scala

```diff
@@ -42,8 +41,7 @@ object HiveSerDe {
         HiveSerDe(
           inputFormat = Option("org.apache.hadoop.hive.ql.io.RCFileInputFormat"),
           outputFormat = Option("org.apache.hadoop.hive.ql.io.RCFileOutputFormat"),
-          serde = Option(conf.getConfString("hive.default.rcfile.serde",
-            "org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnarSerDe"))),
+          serde = Option("org.apache.hadoop.hive.serde2.columnar.LazyBinaryColumnarSerDe")),
```

Comment:
I think as long as we can read RCFile data serialized with LazySimpleColumnarSerDe. We are good. When creating a new RCFile, I think the binary serde needs to be used. Btw, we need to a release note to mention this change.